### ISSUE TITLE
✨ Support enhanced vertical ruler functionality / customization

### ISSUE THREAD

**Issue by ben-p-commits** [CodeEditApp/CodeEditTextView#127]
## Description

The vertical ruler (`NSRulerView`) has a number of requirements that no longer can be efficiently met by STTextView's implementation.  

The recommendation is for an NSRulerView subclass to instead be encapsulated within CodeEditTextView, to provide easier workflows for contributors to customize and enhance it. See the bottom of this description for a list of proposed enhancements after this port is complete.

The goal of this enhancement should be to provide basic RulerView functionality aligned with what exists today in the project. **It is not to implement the enhancements listed below!**

**Future enhancements will likely include**
- support for the same preferences available in Xcode's preferences: 
  <img width="234" alt="image" src="https://user-images.githubusercontent.com/5067237/214606431-b1b4205b-49ed-4d8a-84eb-5abac41b7393.png">
- toggle line numbers on/off without hiding other RulerView features
- A code folding ribbon
  <img width="64" alt="image" src="https://user-images.githubusercontent.com/5067237/214607409-00bfbdc5-e4d3-4899-a915-f3c17942a6ce.png">
- Code comparison controls
  <img width="100" alt="image" src="https://user-images.githubusercontent.com/5067237/214607223-78bce777-0236-41e7-a345-c5f9c44bc1d9.png">

**Comment by austincondiff** [comment]
Is this issue still necessary now that we are using our own text view?

**Comment by thecoolwinter** [comment]
Yeah this is our gutter view. We ended up going a slightly different direction and used a floating subview over an `NSRulerView` but it's the same.
